clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockInputStream.java

```diff
@@ -142,22 +168,19 @@ private void readBlock() throws IOException {
 
         // verify checksum
         if (flg.isBlockChecksumSet() && Utils.readUnsignedIntLE(in) != checksum.hash(bufferToRead, 0, blockSize, 0)) {
-            throw new IOException(BLOCK_HASH_MISMATCH);
+            throw new LZ4Exception(BLOCK_HASH_MISMATCH);
```

Comment:
Before I added the comment above, I checked what the existing `net.jpountz.lz4.LZ4BlockInputStream` does and it has code like:

```java
try {
        final int compressedLen2 = decompressor.decompress(compressedBuffer, 0, buffer, 0, originalLen);
        if (compressedLen != compressedLen2) {
          throw new IOException("Stream is corrupted");
        }
      } catch (LZ4Exception e) {
        throw new IOException("Stream is corrupted", e);
      }
```

So, it doesn't seem clear to me that the upstream `InputStream` and `OutputStream` classes would be throwing `LZ4Exception` which inherits from `RuntimeException` instead of `IOException` or a subclass (that would be unusual). Why do you think it would do that?